linter/src/rules/removing_existing_index.rs

```diff
@@ -0,0 +1,110 @@
+use crate::violations::{RuleViolation, RuleViolationKind};
+use crate::ViolationMessage;
+use squawk_parser::ast::{AlterTableCmds, AlterTableType, ObjectType, RawStmt, Stmt};
+
+#[must_use]
+pub fn removing_existing_index(tree: &[RawStmt]) -> Vec<RuleViolation> {
+  let mut errs = vec![];
+  for raw_stmt in tree {
+    match &raw_stmt.stmt {
+      Stmt::DropStmt(stmt) if stmt.remove_type == ObjectType::Index => {
+        errs.push(RuleViolation::new(
+          RuleViolationKind::RemovingExistingIndex,
+          raw_stmt.into(),
+          Some(vec![
+            ViolationMessage::Note(
+              "Removing an existing index can lead to downtime if the index is heavily used."
+                .into(),
+            ),
+            ViolationMessage::Help(
+              "Ensure that query patterns do not rely on this index before removing.".into(),
+            ),
+          ]),
+        ))
+      }
+      Stmt::AlterTableStmt(stmt) => {
+        for AlterTableCmds::AlterTableCmd(cmd) in &stmt.cmds {
+          if cmd.subtype == AlterTableType::DropConstraint {
+            if String::from(cmd.name.as_ref().unwrap()).contains("pkey") {
```

Comment:
this is not a good check, but there is no `def` for `DropConstraint` which contains the constraint info like `contype == Primary`